golem/resources/agreement/agreement.py

```diff
@@ -52,6 +53,7 @@ async def wait_for_approval(self) -> bool:
         """
         try:
             await self.api.wait_for_approval(self.id, timeout=15, _request_timeout=16)
+            self._approved_at = datetime.now(timezone.utc)
```

Comment:
I'd move this to _before_ the `wait_for_approval` line, so that we're sure our timer is started _before_ the provider's

Reply:
Both comments are on the point, let's do both: Store time before `wait_for_approval` but assign it to `self._approved_at` after success.

Reply:
Changed

Reply:
👍